src/main/java/org/terasology/caves/CaveToDensityProvider.java

```diff
@@ -37,7 +36,8 @@ public void process(GeneratingRegion region) {
         CaveFacet caveFacet = region.getRegionFacet(CaveFacet.class);
         DensityFacet densityFacet = region.getRegionFacet(DensityFacet.class);
 
-        for (Vector3i pos : region.getRegion()) {
+        for (Vector3ic position : region.getRegion()) {
+            Vector3i pos = new Vector3i(position);
```

Comment:
No need to create a new vector here, should be fine to use the constant vector directly:
```suggestion
        for (Vector3ic pos : region.getRegion()) {
```